gradleutils-shared/src/main/java/net/minecraftforge/gradleutils/shared/SharedUtil.java

```diff
@@ -250,6 +252,129 @@ public List<String> getArgs() {
 
     //region Dependency Information
 
+    public record SimpleModuleIdentifier(String getGroup, String getName) implements ModuleIdentifier {
```

Comment:
My be worth stealing [this](https://github.com/MinecraftForge/MinecraftMavenizer/blob/master/src/main/java/net/minecraftforge/mcmaven/impl/util/Artifact.java) its tried and tested.

Reply:
`artifact-utils` :P